src/NuGet.Core/NuGet.Build.Tasks/NuGet.Build.Tasks.Pack.targets

```diff
@@ -14,8 +14,8 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <!-- Load NuGet.Build.Tasks.Pack.dll, this can be overridden to use a different version with $(NuGetPackTaskAssemblyFile) -->
   <PropertyGroup Condition="$(NuGetPackTaskAssemblyFile) == ''">
-    <NuGetPackTaskAssemblyFile Condition="'$(MSBuildRuntimeType)' == 'Core'">..\CoreCLR\NuGet.Build.Tasks.Pack.dll</NuGetPackTaskAssemblyFile>
-    <NuGetPackTaskAssemblyFile Condition="'$(MSBuildRuntimeType)' != 'Core'">..\Desktop\NuGet.Build.Tasks.Pack.dll</NuGetPackTaskAssemblyFile>
+    <NuGetPackTaskAssemblyFile Condition="'$(UsingMicrosoftNetSdk)' == 'true' And '$(MSBuildRuntimeType)' != 'Core'">Sdks\Microsoft.NET.Sdk\tools\net472\NuGet.Build.Tasks.Pack.dll</NuGetPackTaskAssemblyFile>
```

Comment:
nit
```suggestion
    <NuGetPackTaskAssemblyFile Condition="'$(UsingMicrosoftNetSdk)' == 'true' AND '$(MSBuildRuntimeType)' != 'Core'">Sdks\Microsoft.NET.Sdk\tools\net472\NuGet.Build.Tasks.Pack.dll</NuGetPackTaskAssemblyFile>
```

Reply:
> `$(MSBuildRuntimeType)' != 'Core'`

Why is this changing from Core type to not Core type when the .NET SDK is used?
Does it now mean, only add the assembly for Mono & .NET Framework?

Reply:
Previously the two dlls were in relative directories:
```text
..\CoreCLR\NuGet.Build.Tasks.Pack.dll
..\Desktop\NuGet.Build.Tasks.Pack.dll
```

So, the path to the assembly only changed based on dotnet cli or msbuild.

Once the linked issue is complete (NuGet, the .NET SDK, and VS are all updated), VS is only going to ship the net472 dll, and it will be in the same directory as the targets file.  The .NET SDK will ship the net9.0 dll in the same directory as the targets file, and will ship the net472 dll in `Sdks\Microsoft.NET.Sdk\tools\net472\NuGet.Build.Tasks.Pack.dll` for when customers use msbuild.exe to run pack on an SDK style project.

Reply:
> nit

regarding the suggestion, if you look near the bottom of the first table in these docs: https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild-conditions?view=vs-2022

You'll see that MSBuild's docs use `And` as the documented notation.  MSBuild is case insenstive, so all permuations of `and`, `AND`, `aNd`, and so on work.

I remember someone making the same suggestion to Jeff on one of his PRs, and this was his justification for why to use `And`.